tpm2/src/util/tpm2_kmyth_io.c

```diff
@@ -1101,3 +1618,36 @@ int print_to_stdout(unsigned char *data, size_t data_size)
   BIO_free_all(bdata);
   return 0;
 }
+
+int concat(uint8_t ** dest, size_t *dest_length, uint8_t * input,
+           size_t input_length)
+{
+  if (input == NULL || input_length == 0) //nothing to concat
+  {
+    return (0);
+  }
+
+  uint8_t *new_dest = NULL;
+  size_t new_dest_len = *dest_length + input_length;
+  size_t offset = *dest_length;
+
+  if (new_dest_len < *dest_length)  //if we have an overflow
+  {
+    kmyth_log(LOG_ERR, "Maximum array size exceeded ... exiting");
+    return (1);
+  }
+
+  if ((new_dest = realloc(*dest, new_dest_len)) == 0)
+  {
+    kmyth_log(LOG_ERR, "Ran out of memory ... exiting");
+    return (1);
+  }
+
+  for (size_t i = 0; i < input_length; i++)
```

Comment:
It would be cleaner to do this with a memcpy rather than iterating.

Reply:
Agreed and done.